xilem/src/view/board.rs

```diff
@@ -0,0 +1,384 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use std::marker::PhantomData;
+
+use masonry::{
+    widget::{self, KurboShape, PositionedElement, SvgElement, WidgetMut},
+    Widget,
+};
+use xilem_core::{
+    AnyElement, AnyView, AppendVec, DynMessage, ElementSplice, MessageResult, Mut, SuperElement,
+    View, ViewElement, ViewMarker, ViewSequence,
+};
+
+use crate::{Pod, ViewCtx, WidgetView};
+
+pub use masonry::widget::BoardParams;
+
+mod kurbo_shape;
+mod style_modifier;
+
+pub use kurbo_shape::GraphicsView;
+pub use style_modifier::{fill, stroke, transform, Fill, GraphicsExt, Stroke, Transform};
+
+pub fn board<State, Action, Seq: BoardSequence<State, Action>>(
+    sequence: Seq,
+) -> Board<Seq, State, Action> {
+    Board {
+        sequence,
+        phantom: PhantomData,
+    }
+}
+
+pub struct Board<Seq, State, Action = ()> {
+    sequence: Seq,
+    phantom: PhantomData<fn() -> (State, Action)>,
+}
+
+impl<Seq, State, Action> ViewMarker for Board<Seq, State, Action> {}
+impl<State, Action, Seq> View<State, Action, ViewCtx> for Board<Seq, State, Action>
+where
+    State: 'static,
+    Action: 'static,
+    Seq: BoardSequence<State, Action>,
+{
+    type Element = Pod<widget::Board>;
+
+    type ViewState = Seq::SeqState;
+
+    fn build(&self, ctx: &mut ViewCtx) -> (Self::Element, Self::ViewState) {
+        let mut elements = AppendVec::default();
+        let mut widget = widget::Board::new();
+        let seq_state = self.sequence.seq_build(ctx, &mut elements);
+        for BoardElement { element } in elements.into_inner() {
+            widget = widget.with_child_pod(element.inner);
+        }
+        (Pod::new(widget), seq_state)
+    }
+
+    fn rebuild<'el>(
+        &self,
+        prev: &Self,
+        view_state: &mut Self::ViewState,
+        ctx: &mut ViewCtx,
+        element: Mut<'el, Self::Element>,
+    ) -> Mut<'el, Self::Element> {
+        // TODO: Re-use scratch space?
+        let mut splice = BoardSplice::new(element);
+        self.sequence
+            .seq_rebuild(&prev.sequence, view_state, ctx, &mut splice);
+        debug_assert!(splice.scratch.is_empty());
+        splice.element
+    }
+
+    fn teardown(
+        &self,
+        view_state: &mut Self::ViewState,
+        ctx: &mut ViewCtx,
+        element: Mut<'_, Self::Element>,
+    ) {
+        let mut splice = BoardSplice::new(element);
+        self.sequence.seq_teardown(view_state, ctx, &mut splice);
+        debug_assert!(splice.scratch.into_inner().is_empty());
+    }
+
+    fn message(
+        &self,
+        view_state: &mut Self::ViewState,
+        id_path: &[xilem_core::ViewId],
+        message: DynMessage,
+        app_state: &mut State,
+    ) -> MessageResult<Action> {
+        self.sequence
+            .seq_message(view_state, id_path, message, app_state)
+    }
+}
+
+pub type AnyBoardView<State, Action = ()> =
+    dyn AnyView<State, Action, ViewCtx, BoardElement> + Send + Sync;
+
+pub struct BoardElement {
```

Comment:
I don't think this extra type is necessary.
I think you can just use `Pod<W: SvgElement>`.
And the `AnyView` could be (see below for more info)

```rust
pub type AnySvgView<State, Action = ()> =
    dyn AnyView<State, Action, ViewCtx, Pod<DynWidget<Box<dyn SvgElement>>>> + Send + Sync;